### ISSUE TITLE
Err-6:CL_OUT_OF_HOST_MEMORY  when running QuickVina-W-GPU.

### ISSUE THREAD

**Issue by yanbosmu** [DeltaGroupNJUPT/Vina-GPU-2.1#17]
I only have problem in QuickVina-W-GPU.
it always shows Err-6:CL_OUT_OF_HOST_MEMORY.
setting Stack Reserve Size into at least 7,000,000 or unlimited didn't work. Is that ulimit -s ulimit or 7000000?
can you help me fix it?

**Comment by thezsd** [comment]
> I only have problem in QuickVina-W-GPU. it always shows Err-6:CL_OUT_OF_HOST_MEMORY. setting Stack Reserve Size into at least 7,000,000 or unlimited didn't work. Is that ulimit -s ulimit or 7000000? can you help me fix it?
I'm having the same problem, I still have 10g of video memory on my card but still have the problem, have you solved it yet?

**Comment by yanbosmu** [comment]
> > I only have problem in QuickVina-W-GPU. it always shows Err-6:CL_OUT_OF_HOST_MEMORY. setting Stack Reserve Size into at least 7,000,000 or unlimited didn't work. Is that ulimit -s ulimit or 7000000? can you help me fix it?
> 
> I'm having the same problem, I still have 10g of video memory on my card but still have the problem, have you solved it yet?

NO, I gave up using Quickvina-w, I think it was due to the limited memory rom in GPU card.

**Comment by Glinttsd** [comment]
Hi, we have checked and fixed this issue in our commit ba30b25ef25b4be10fe489d485a90f908684b751, please check it out

**Comment by duanxiaoqian** [comment]
> Hi, we have checked and fixed this issue in our commit [ba30b25](https://github.com/DeltaGroupNJUPT/Vina-GPU-2.1/commit/ba30b25ef25b4be10fe489d485a90f908684b751), please check it out

i'm having the same problem , my code have changed following the commit [ba30b25](https://github.com/DeltaGroupNJUPT/Vina-GPU-2.1/commit/ba30b25ef25b4be10fe489d485a90f908684b751), but not work~

**Comment by vinays6** [comment]
Still experiencing this error on the basic example given in the newest version. Running `ulimit -s 8192` or even `ulimit -s 7000000` doesn't change anything. 
@Glinttsd Any ideas?

**Comment by MFSAWUY** [comment]
I've run into the same issue. As mentioned in https://github.com/DeltaGroupNJUPT/Vina-GPU-2.1/issues/49, it seems to be related to calling the compiler from within OpenCL, which ends up leaking a ton of memory.

**Comment by vinays6** [comment]
Spent a really long time debugging but I eventually got it. The issue is that this line https://github.com/DeltaGroupNJUPT/Vina-GPU-2.1/blob/180272b8a5265d6ed9664178345933cebe2cd349/QuickVina-W-GPU-2.1/OpenCL/src/kernels/kernel2.cl#L79 takes up huge amounts of memory, much more than is needed. I've put this container list on the host and it does work (yay!). I'll fork the repo for all those interested.

**Comment by xrik** [comment]
> Spent a really long time debugging but I eventually got it. The issue is that this line
> 
> [Vina-GPU-2.1/QuickVina-W-GPU-2.1/OpenCL/src/kernels/kernel2.cl](https://github.com/DeltaGroupNJUPT/Vina-GPU-2.1/blob/180272b8a5265d6ed9664178345933cebe2cd349/QuickVina-W-GPU-2.1/OpenCL/src/kernels/kernel2.cl#L79)
> 
> Line 79 in [180272b](/DeltaGroupNJUPT/Vina-GPU-2.1/commit/180272b8a5265d6ed9664178345933cebe2cd349)
> 
>  __private individual_container list; 
> takes up huge amounts of memory, much more than is needed. I've put this container list on the host and it does work (yay!). I'll fork the repo for all those interested.

Hi vinays6:
i managed to compile your pull with large box and run once successfully, and then run the second one, i got those "
Kernel2: g_evaluate ERROR!#5
Kernel2: g_evaluate ERROR!#5
Kernel2: g_evaluate ERROR!#5
kernel2: g_evaluate ERROR!#1
kernel2: g_evaluate ERROR!#1
kernel2: g_evaluate ERROR!#1
kernel2: g_evaluate ERROR!#1
" message with the same program i have run successfully, do you have such problem?
Thanks a lot!

**Comment by vinays6** [comment]
I'm not sure why that is. I haven't been able to successfully use large_box as I don't have a machine with the required GPU memory yet.
